metricplugin/metric_export.go

```diff
@@ -264,14 +275,23 @@ func (mep *MetricExporterPlugin) populatePrometheus(interval time.Duration) {
 			av := agentVersion.(string)
 			countsByAgentVersion[av] = countsByAgentVersion[av] + 1
 		}
+		// Sort the agent versions in decreasing order
+		sortedAgentVersions := make([]string, 0, len(countsByAgentVersion))
+		for av := range countsByAgentVersion {
+			sortedAgentVersions = append(sortedAgentVersions, av)
+		}
+		sort.Slice(sortedAgentVersions, func(i, j int) bool {
+			// This expects a Less() function, but since we want descending order we use ">"
+			return countsByAgentVersion[sortedAgentVersions[i]] > countsByAgentVersion[sortedAgentVersions[j]]
+		})
 		elapsed = time.Since(before)
 
 		log.Debugf("populatePrometheus: took %s to calculate agent version counts %+v", elapsed, countsByAgentVersion)
 		agentVersionCount.Reset()
-		for agentVersion, count := range countsByAgentVersion {
+		for _, agentVersion := range sortedAgentVersions[0:mep.conf.AgentVersionCutOff] {
```

Comment:
You can omit the lower bound, zero is implicit: `sortedAgentVersions[:mep.conf.AgentVersionCutOff]`